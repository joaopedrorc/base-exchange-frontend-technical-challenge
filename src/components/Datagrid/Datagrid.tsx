'use client';

import {
  ColumnDef,
  ColumnFiltersState,
  SortingState,
  flexRender,
  getCoreRowModel,
  getPaginationRowModel,
  getFilteredRowModel,
  useReactTable,
  getSortedRowModel,
} from '@tanstack/react-table';

import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from '@/components/ui/table';
import { Button } from '../ui/button';
import { useState } from 'react';
import { Input } from '../ui/input';
import { Label } from '@radix-ui/react-label';

interface DatagridProps<TData, TValue> {
  columns: ColumnDef<TData, TValue>[];
  data: TData[];
}

export function Datagrid<TData, TValue>({
  columns,
  data,
}: DatagridProps<TData, TValue>) {
  const [sorting, setSorting] = useState<SortingState>([]);
  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
  const [selectedRow, setSelectedRow] = useState<TData | null>(null);

  const table = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
    onSortingChange: setSorting,
    getSortedRowModel: getSortedRowModel(),
    onColumnFiltersChange: setColumnFilters,
    getFilteredRowModel: getFilteredRowModel(),
    state: {
      sorting,
      columnFilters,
    },
  });

  // Send to Order details page and fetch all details from that order

  return (
    <div>
      <h2 className="bold text-2xl">Filtros</h2>

      <div className="flex flex-col items-center gap-4 py-4 md:flex-row">
        <div className="max-w-3x flex w-full flex-col py-2 md:max-w-3xs">
          <Label htmlFor="id" className="text-xs">
            Por ID
          </Label>

          <Input
            name="id"
            className="max-w-full"
            placeholder="Digite um ID"
            value={(table.getColumn('id')?.getFilterValue() as string) ?? ''}
            onChange={(event) =>
              table.getColumn('id')?.setFilterValue(event.target.value)
            }
          />
        </div>

        <div className="flex w-full flex-col py-2 md:max-w-3xs">
          <Label htmlFor="asset" className="text-xs">
            Por ação
          </Label>

          <Input
            name="asset"
            className="max-w-full md:max-w-3xs"
            placeholder="Digite uma ação"
            value={(table.getColumn('asset')?.getFilterValue() as string) ?? ''}
            onChange={(event) =>
              table.getColumn('asset')?.setFilterValue(event.target.value)
            }
          />
        </div>

        <div className="flex w-full flex-col py-2 md:max-w-3xs">
          <Label htmlFor="status" className="text-xs">
            Por status
          </Label>

          <Input
            name="status"
            className="max-w-full md:max-w-3xs"
            placeholder="Digite um status"
            value={
              (table.getColumn('status')?.getFilterValue() as string) ?? ''
            }
            onChange={(event) =>
              table.getColumn('status')?.setFilterValue(event.target.value)
            }
          />
        </div>

        <div className="flex w-full flex-col py-2 md:max-w-3xs">
          <Label htmlFor="date" className="text-xs">
            Por data
          </Label>

          <Input
            name="asset"
            className="max-w-full md:max-w-3xs"
            placeholder="Digite uma data"
            value={
              (table.getColumn('dateAndTime')?.getFilterValue() as string) ?? ''
            }
            onChange={(event) =>
              table.getColumn('dateAndTime')?.setFilterValue(event.target.value)
            }
          />
        </div>

        <div className="flex w-full flex-col py-2 md:max-w-3xs">
          <Label htmlFor="side" className="text-xs">
            Por operação
          </Label>

          <Input
            name="side"
            className="max-w-full md:max-w-3xs"
            placeholder="Digite uma operação"
            value={(table.getColumn('side')?.getFilterValue() as string) ?? ''}
            onChange={(event) =>
              table.getColumn('side')?.setFilterValue(event.target.value)
            }
          />
        </div>
      </div>

      <h2 className="bold mb-4 text-2xl">Tabela de operações</h2>

      <div className="text-muted-foreground flex-1 text-sm">
        Selecione uma operação clicando em cima da linha desejada.
      </div>

      <div className="rounded-md border">
        <Table>
          <TableHeader>
            {table.getHeaderGroups().map((headerGroup) => (
              <TableRow key={headerGroup.id}>
                {headerGroup.headers.map((header) => {
                  return (
                    <TableHead key={header.id}>
                      {header.isPlaceholder
                        ? null
                        : flexRender(
                            header.column.columnDef.header,
                            header.getContext()
                          )}
                    </TableHead>
                  );
                })}
              </TableRow>
            ))}
          </TableHeader>
          <TableBody>
            {table.getRowModel().rows?.length ? (
              table.getRowModel().rows.map((row) => {
                const isSelected = selectedRow === row.original;
                return (
                  <TableRow
                    key={row.id}
                    data-state={row.getIsSelected() && 'selected'}
                    className={
                      isSelected
                        ? 'cursor-pointer bg-slate-800'
                        : 'cursor-pointer'
                    }
                    onClick={() => setSelectedRow(row.original)}
                  >
                    {row.getVisibleCells().map((cell) => (
                      <TableCell
                        key={cell.id}
                        className={isSelected ? 'text-white' : 'text-oklab'}
                      >
                        {flexRender(
                          cell.column.columnDef.cell,
                          cell.getContext()
                        )}
                      </TableCell>
                    ))}
                  </TableRow>
                );
              })
            ) : (
              <TableRow>
                <TableCell
                  colSpan={columns.length}
                  className="h-24 text-center"
                >
                  Sem resultados.
                </TableCell>
              </TableRow>
            )}
          </TableBody>
        </Table>
      </div>
      <div className="flex items-center justify-end space-x-2 py-4">
        <Button
          variant="outline"
          size="sm"
          onClick={() => table.previousPage()}
          disabled={!table.getCanPreviousPage()}
        >
          Anterior
        </Button>
        <Button
          variant="outline"
          size="sm"
          onClick={() => table.nextPage()}
          disabled={!table.getCanNextPage()}
        >
          Próxima
        </Button>
      </div>
    </div>
  );
}
