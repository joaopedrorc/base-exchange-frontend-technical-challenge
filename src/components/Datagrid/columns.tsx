import { ColumnDef } from '@tanstack/react-table';

export type Order = {
  id: string;
  asset: string;
  side: string;
  price: string;
  amount: number;
  amountLeft: number;
  status: 'Aberta' | 'Parcial' | 'Executada' | 'Cancelada';
  dateAndTime: string;
};

export const columns: ColumnDef<Order>[] = [
  {
    accessorKey: 'id',
    header: 'ID',
  },
  {
    accessorKey: 'asset',
    header: 'Ação',
  },
  {
    accessorKey: 'side',
    header: 'Operação',
  },
  {
    accessorKey: 'price',
    header: 'Preço',
  },
  {
    accessorKey: 'amount',
    header: 'Quantidade',
  },
  {
    accessorKey: 'amountLeft',
    header: 'Restante',
  },
  {
    accessorKey: 'status',
    header: 'Status',
  },
  {
    accessorKey: 'dateAndTime',
    header: 'Data/Hora',
  },
];
