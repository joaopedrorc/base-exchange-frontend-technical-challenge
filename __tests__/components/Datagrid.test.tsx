import { render } from '@testing-library/react';
import { Datagrid } from '@/components/Datagrid/Datagrid';

const dataMock = [
  {
    id: 'ORD-1',
    asset: 'PETR4',
    side: 'Compra',
    price: '28,45',
    amount: 100,
    amountLeft: 100,
    status: 'Aberta',
    dateAndTime: '24/04/2025, 09:15:32',
    history: ['Pedido criado em 24/04/2025, 09:15:32'],
  },
];

const mockColumns = [
  {
    accessorKey: 'id',
    header: 'ID',
  },
  {
    accessorKey: 'asset',
    header: 'Ação',
  },
  {
    accessorKey: 'side',
    header: 'Operação',
  },
  {
    accessorKey: 'price',
    header: 'Preço',
  },
  {
    accessorKey: 'amount',
    header: 'Quantidade',
  },
  {
    accessorKey: 'amountLeft',
    header: 'Restante',
  },
  {
    accessorKey: 'status',
    header: 'Status',
  },
  {
    accessorKey: 'dateAndTime',
    header: 'Data/Hora',
  },
];

jest.mock('next/navigation', () => {
  return {
    useRouter: () => ({
      push: () => {},
    }),
  };
});

describe('Datagrid component', () => {
  it('should render Datagrid component', () => {
    render(<Datagrid columns={mockColumns} data={dataMock} />);
  });

  it('renders Datagrid component unchanged', async () => {
    const component = render(
      <Datagrid columns={mockColumns} data={dataMock} />
    );

    expect(component).toMatchSnapshot();
  });
});
