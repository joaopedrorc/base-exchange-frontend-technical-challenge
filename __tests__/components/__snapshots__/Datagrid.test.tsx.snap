// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datagrid component renders Datagrid component unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="bold text-2xl"
        >
          Filtros
        </h2>
        <div
          class="flex flex-col items-center gap-4 py-4 md:flex-row"
        >
          <div
            class="max-w-3x flex w-full flex-col py-2 md:max-w-3xs"
          >
            <label
              class="text-xs"
              for="id"
            >
              Por ID
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full"
              data-slot="input"
              name="id"
              placeholder="Digite um ID"
              value=""
            />
          </div>
          <div
            class="flex w-full flex-col py-2 md:max-w-3xs"
          >
            <label
              class="text-xs"
              for="asset"
            >
              Por ação
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
              data-slot="input"
              name="asset"
              placeholder="Digite uma ação"
              value=""
            />
          </div>
          <div
            class="flex w-full flex-col py-2 md:max-w-3xs"
          >
            <label
              class="text-xs"
              for="status"
            >
              Por status
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
              data-slot="input"
              name="status"
              placeholder="Digite um status"
              value=""
            />
          </div>
          <div
            class="flex w-full flex-col py-2 md:max-w-3xs"
          >
            <label
              class="text-xs"
              for="date"
            >
              Por data
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
              data-slot="input"
              name="asset"
              placeholder="Digite uma data"
              value=""
            />
          </div>
          <div
            class="flex w-full flex-col py-2 md:max-w-3xs"
          >
            <label
              class="text-xs"
              for="side"
            >
              Por operação
            </label>
            <input
              class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
              data-slot="input"
              name="side"
              placeholder="Digite uma operação"
              value=""
            />
          </div>
        </div>
        <h2
          class="bold mb-4 text-2xl"
        >
          Tabela de operações
        </h2>
        <div
          class="text-muted-foreground flex-1 text-sm"
        >
          Selecione uma operação clicando em cima da linha desejada.
        </div>
        <div
          class="rounded-md border"
        >
          <div
            class="relative w-full overflow-x-auto"
            data-slot="table-container"
          >
            <table
              class="w-full caption-bottom text-sm"
              data-slot="table"
            >
              <thead
                class="[&_tr]:border-b"
                data-slot="table-header"
              >
                <tr
                  class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                  data-slot="table-row"
                >
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    ID
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Ação
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Operação
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Preço
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Quantidade
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Restante
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Status
                  </th>
                  <th
                    class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                    data-slot="table-head"
                  >
                    Data/Hora
                  </th>
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
                data-slot="table-body"
              >
                <tr
                  class="data-[state=selected]:bg-muted border-b transition-colors cursor-pointer hover:bg-slate-800 hover:text-white"
                  data-slot="table-row"
                  data-state="false"
                >
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    ORD-1
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    PETR4
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    Compra
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    28,45
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    100
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    100
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    Aberta
                  </td>
                  <td
                    class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                    data-slot="table-cell"
                  >
                    24/04/2025, 09:15:32
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="flex items-center justify-end space-x-2 py-4"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
            data-slot="button"
            disabled=""
          >
            Anterior
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
            data-slot="button"
            disabled=""
          >
            Próxima
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="bold text-2xl"
      >
        Filtros
      </h2>
      <div
        class="flex flex-col items-center gap-4 py-4 md:flex-row"
      >
        <div
          class="max-w-3x flex w-full flex-col py-2 md:max-w-3xs"
        >
          <label
            class="text-xs"
            for="id"
          >
            Por ID
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full"
            data-slot="input"
            name="id"
            placeholder="Digite um ID"
            value=""
          />
        </div>
        <div
          class="flex w-full flex-col py-2 md:max-w-3xs"
        >
          <label
            class="text-xs"
            for="asset"
          >
            Por ação
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
            data-slot="input"
            name="asset"
            placeholder="Digite uma ação"
            value=""
          />
        </div>
        <div
          class="flex w-full flex-col py-2 md:max-w-3xs"
        >
          <label
            class="text-xs"
            for="status"
          >
            Por status
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
            data-slot="input"
            name="status"
            placeholder="Digite um status"
            value=""
          />
        </div>
        <div
          class="flex w-full flex-col py-2 md:max-w-3xs"
        >
          <label
            class="text-xs"
            for="date"
          >
            Por data
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
            data-slot="input"
            name="asset"
            placeholder="Digite uma data"
            value=""
          />
        </div>
        <div
          class="flex w-full flex-col py-2 md:max-w-3xs"
        >
          <label
            class="text-xs"
            for="side"
          >
            Por operação
          </label>
          <input
            class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive max-w-full md:max-w-3xs"
            data-slot="input"
            name="side"
            placeholder="Digite uma operação"
            value=""
          />
        </div>
      </div>
      <h2
        class="bold mb-4 text-2xl"
      >
        Tabela de operações
      </h2>
      <div
        class="text-muted-foreground flex-1 text-sm"
      >
        Selecione uma operação clicando em cima da linha desejada.
      </div>
      <div
        class="rounded-md border"
      >
        <div
          class="relative w-full overflow-x-auto"
          data-slot="table-container"
        >
          <table
            class="w-full caption-bottom text-sm"
            data-slot="table"
          >
            <thead
              class="[&_tr]:border-b"
              data-slot="table-header"
            >
              <tr
                class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
                data-slot="table-row"
              >
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  ID
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Ação
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Operação
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Preço
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Quantidade
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Restante
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Status
                </th>
                <th
                  class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  data-slot="table-head"
                >
                  Data/Hora
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
              data-slot="table-body"
            >
              <tr
                class="data-[state=selected]:bg-muted border-b transition-colors cursor-pointer hover:bg-slate-800 hover:text-white"
                data-slot="table-row"
                data-state="false"
              >
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  ORD-1
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  PETR4
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  Compra
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  28,45
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  100
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  100
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  Aberta
                </td>
                <td
                  class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-oklab"
                  data-slot="table-cell"
                >
                  24/04/2025, 09:15:32
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="flex items-center justify-end space-x-2 py-4"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
          data-slot="button"
          disabled=""
        >
          Anterior
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
          data-slot="button"
          disabled=""
        >
          Próxima
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
